### PR TITLE
Added F0 boards

### PR DESCRIPTION
**Pull request by alextrical**
| :yellow_heart: | STM32F030C6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F030K6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F031C4<br>STM32F031C6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F031E6<br>STM32F038E6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F031F4<br>STM32F031F6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F031G4<br>STM32F031G6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F031K4<br>STM32F031K6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F038C6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F038F6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F038G6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F038K6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F042F4<br>STM32F042F6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F042G4<br>STM32F042G6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F042T6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F048G6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F048T6 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F051C4 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F051K4 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F051R4 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F051T8 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F058C8 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F058R8 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F058T8 | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F071C8<br>STM32F071CB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F071RB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F071V8<br>STM32F071VB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F072V8<br>STM32F072VB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F078CB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F078RB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F078VB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F091CB<br>STM32F091CC | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F091VB | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F091VC | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F098CC | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F098RC | Generic Board | **2.3.0** |  |
| :yellow_heart: | STM32F098VC | Generic Board | **2.3.0** |  |

### CODE REVIEW

File: variants/STM32F0xx/F030C6T/STM32F030C6TX_ldscript.ld [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+_estack = ORIGIN(RAM) + LENGTH(RAM); /* end of "RAM" Ram type memory */
+
+_Min_Heap_Size = 0x200 ; /* required amount of heap */
+_Min_Stack_Size = 0x400 ; /* required amount of stack */
+
+/* Memories definition */
+MEMORY
+{
+  RAM    xrw    : ORIGIN = 0x20000000,   LENGTH = LD_MAX_DATA_SIZE
+  FLASH    rx    : ORIGIN = 0x8000000 + LD_FLASH_OFFSET, LENGTH = LD_MAX_SIZE - LD_FLASH_OFFSET
```
Comment (ABOSTM) [inline variants/STM32F0xx/F030C6T/STM32F030C6TX_ldscript.ld:48]: CI check fails (compilation) due to missing parenthesis. Apply to all *.ld files
```suggestion
  RAM    (xrw)    : ORIGIN = 0x20000000,   LENGTH = LD_MAX_DATA_SIZE
  FLASH    (rx)    : ORIGIN = 0x8000000 + LD_FLASH_OFFSET, LENGTH = LD_MAX_SIZE - LD_FLASH_OFFSET
```

**Review by ABOSTM — CHANGES_REQUESTED** [review]
*.ld file to be renamed

File: variants/STM32F0xx/F030C6T/STM32F030C6TX_ldscript.ld (hunk 1 above)
Comment (alextrical) [reply variants/STM32F0xx/F030C6T/STM32F030C6TX_ldscript.ld]: Added parenthesis back into files, incorrect RegEx obliterated them

*.ld files renamed to correct names

**Review by ABOSTM — CHANGES_REQUESTED** [review]
*2.4.0*

**Comment by ABOSTM** [comment]
CI failed is linked to issue #1798, we will take care of it.

**Comment by ABOSTM** [comment]
CI failed is linked to issue #1798, we will take care of it.

File: README.md [hunk 2]
```diff
@@ -308,6 +345,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :green_heart: | STM32G031J4<br>STM32G031J6 | Generic Board | *2.0.0* |  |
 | :green_heart: | STM32G031F4<br>STM32G031F6<br>STM32G031F8 | Generic Board | *2.3.0* |  |
 | :green_heart: | STM32G031K4<br>STM32G031K6<br>STM32G031K8 | Generic Board | *2.0.0* |  |
+| :green_heart: | STM32G031K4<br>STM32G031K6<br>STM32G031K8 | Generic Board | *2.0.0* |  |
```
Comment (ABOSTM) [inline README.md:348]: Duplicated line.
To be removed.

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (alextrical) [reply README.md]: removed the requested line.

**Review by fpistm — APPROVED** [review]
LGTM

**Pull request merged**